server/client.go

```diff
@@ -610,12 +610,15 @@ func (c *client) initClient() {
 			if conn = addr.String(); conn != _EMPTY_ {
 				host, port, _ := net.SplitHostPort(conn)
 				iPort, _ := strconv.Atoi(port)
-				c.port = uint16(iPort)
+				c.host, c.port = host, uint16(iPort)
 				if c.isWebsocket() && c.ws.clientIP != _EMPTY_ {
-					c.host = c.ws.clientIP
-					conn = net.JoinHostPort(c.host, port)
-				} else {
-					c.host = host
+					cip := c.ws.clientIP
+					// Surround IPv6 addresses with square brackets, as
+					// net.JoinHostPort would do...
+					if strings.Contains(cip, ":") {
+						cip = "[" + cip + "]"
+					}
+					conn = fmt.Sprintf("%s/%s", cip, conn)
```

Comment:
Do we normally have reflection bounces inside the client connection execution path?

Reply:
Sorry, I am not sure what you are asking here. Is it about the use of fmt.Sprintf() (which we use also down below in original code)? The intent of the code above is to surround an IPv6 address with square brackets. The net.JoinHostAndPort() does exactly that.

Reply:
Yes.  I approved the PR, it's fine to merge, I was just pointing at this in case it's an issue so you could adjust it if desired.